cowprotocol/cowswap
# Release/2024.01.23

# Summary




Includes:
59aa84f6b Merge branch 'main' into develop
c20ee4a70 feat(appzi): add appzi triggers for limit orders (#3676)
06ca09d6a chore: address various linter issues (#3677)
72526eb96 fix(fee=0): buy order amounts again (#3679)
2b20e2f50 Merge pull request #3673 from cowprotocol/main
66582ec8c fix(swap): display input amount usd value even if there is no quote (#3658)
ae2eda5d3 feat: fix twitter icon (#3668)
fc69df4d6 feat: update cow-sdk to 4.1.0 (#3660)
b1deee276 fix: fix order display amounts (#3657)
34af83798 feat: animate settings icon (#3665)
a93c27c19 fix(explorer): fix app-data encode form error displaying (#3659)
d41a7a9a5 feat: update Twitter-X menu icon (#3663)
1536727af feat: fix custom recipient style (#3661)
25c7280fc feat: menu text underline style (#3662)
6b4aec988 refactor(explorer): use common networks configs (#3633)
68e09c996 refactor(cowswap): clean up chains configs and rename consts (#3632)
0e13138ce fix(twap): display TWAP instead of Limit in order receipt (#3627)
9c9c2f624 refactor(explorer): remove unused code (#3626)
101ebb5de Merge pull request #3656 from cowprotocol/main
8dfd5add6 Merge pull request #3637 from cowprotocol/main
00157ca8d refactor: remove some explorer dead code, non-exhaustive (#3625)


# To Test
Tests all main flows, special attention to 👀:
- appZi poping up for limit orders
- Buy orders!! (before the test later today)
- Make a tiny order that the fee is bigger than sell amount, and see the USD estimation
- Special attention to the buy/sell amounts you sign

### Permit
- [x] On mainnet, sell a permittable token without previous approval (SWAP/LIMIT)
- [x] Permittable tokens should have a flag in the token selector (SWAP/LIMIT)
- [x] Permittable should NOT be used on TWAP
- [x] Permittable should NOT be used for SC wallets
- [x] Add a permittalbe token not in the list and have it recognized and traded as such. Example `0x30d20208d987713f46dfd34ef128bb16c404d10f`

## Regular flow

Smart contract wallet (Safe):
- [x] Just swap
- [x] Approve + swap
- [x] Wrap + approve + swap (eth-flow)
- [x] Wrap/unwrap
- [x] EthFlow swap
- [x] EthFlow cancellation
- [x] Order placement and fill (Limit/Twap)
- [x] Hard cancel order (Swap/Limit/Twap)

EOA wallet (Metamask):
- [x] Just swap
- [x] Approve + swap
- [x] Wrap + approve + swap (eth-flow)
- [x] Wrap/unwrap
- [x] EthFlow swap
- [x] EthFlow cancellation
- [x] Order placement and fill (Limit/Twap)
- [x] Hard cancel order (Swap/Limit/Twap)
- [x] Soft cancel order (Swap/Limit/Twap)

WalletConnect:
- [x] Just swap
- [ ] Approve + swap
- [ ] Wrap + approve + swap (eth-flow)
- [x] Wrap/unwrap
- [x] EthFlow swap
- [ ] EthFlow cancellation
- [x] Order placement and fill (Limit/Twap)
- [ ] Hard cancel order (Swap/Limit/Twap)
- [x] Soft cancel order (Swap/Limit/Twap)

Account page:
- [x] add token to Metamask
- [x] approve token
- [x] COW balance displaying
- [x] token search by name/address
- [x] add token to Metamask
- [x] favorite token add/remove

Common (WalletConnect/Metamask):
- [x] network switching
- [ ] network persisting between modules
- [x] wallet disconnect
- [x] change wallet
- [x] balances loading
- [x] appzi
- [x] all FAQ pages open
